nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/StreamBulkScroller.java

```diff
@@ -66,52 +64,63 @@
 
     public static final String SCROLL_KEEP_ALIVE_SECONDS_OPT = "scrollKeepAlive";
 
+    public static final String BUCKET_SIZE_OPT = "bucketSize";
```

Comment:
Why defining a processor only for the scroller, this should be a bulk action processor that defines the following topology 
(s is for stream, c for computation)
command (s) -> Scroller (c) -> actions ids (s)
counter (s) -> Counter (c)
So you have at least 2 computations a Scroller and a Counter

Also if we want that computations don't write directly to the KV store in order to support a KV failure, we could add a KVWriter computation responsible to update the command status:
command (s) -> Scroller (c) -> actions ids (s)
Scroller (c) -> status (s)
counter (s) -> Counter (c) -> status (s)
status (s) -> KVWriter

Reply:
The addition of the counter is coming in a separate task: https://jira.nuxeo.com/browse/NXP-25099